src/wallet/scriptpubkeyman.cpp

```diff
@@ -1812,7 +1953,7 @@ std::optional<MigrationData> LegacyScriptPubKeyMan::MigrateToDescriptor()
         WalletDescriptor w_desc(std::move(desc), creation_time, 0, 0, 0);
 
         // Make the DescriptorScriptPubKeyMan and get the scriptPubKeys
-        auto desc_spk_man = std::unique_ptr<DescriptorScriptPubKeyMan>(new DescriptorScriptPubKeyMan(m_storage, w_desc, m_keypool_size));
+        auto desc_spk_man = std::unique_ptr<DescriptorScriptPubKeyMan>(new DescriptorScriptPubKeyMan(m_storage, w_desc, /*keypool_size=*/0));
```

Comment:
Worth adding a comment about the consequences of setting `keypool_size=0`? I'm guessing there are still cases where we generate some pubkeys despite having a zero pool size, otherwise you would have removed some of the code and the for-loop below?

Reply:
I don't think this needs to have any additional explanation. `keypool_size` only matters for ranged descriptors. Non-HD keys do not make ranged descriptors, so the value here is ignored. The loop below is for an entirely different set of descriptors and is unrelated to this.

Reply:
Haven't fully grokked the code yet, so I understand if you'd rather not answer my questions, but if you want:

> I don't think this needs to have any additional explanation. keypool_size only matters for ranged descriptors. Non-HD keys do not make ranged descriptors, so the value here is ignored. 

We're in an HD chain `for`-loop, how is this not involving HD keys?
Surely legacy wallets could be HD wallets.

> The loop below is for an entirely different set of descriptors and is unrelated to this.

I'm referring to the loop over `desc_spks` which is using the pubkeys from the `DescriptorScriptPubKeyMan` that's just been created and comes from `DescriptorScriptPubKeyMan::m_map_script_pub_keys` which is filled out in `DescriptorScriptPubKeyMan::TopUpWithDB()` which is affected by `m_keypool_size`.

Would `m_keypool_size` already always be `0` before this change?

Reply:
The line this comment is on is in the one that goes over the non-HD keys. However, I see that you're confused about the similar line in the loop below that iterates the hd chains.

The `keypool_size` here is a runtime value that depends on the startup options. It's is set when the `DescriptorScriptPubKeyman` is instantiated on loading, and the value is never persisted to disk.

The value that does matter is the `range_end` stored in `WalletDescriptor`. This is persisted to disk. When we are going over the hd chains, we set that `range_end` to match the chain counter stored in the `CHDChain` that is being migrated. This results in a descriptor that will produce exactly the same scriptPubKeys as the original hd chain. We are enforcing that the migrated descriptors produce exactly the same scriptPubKeys (no more, no less). Setting `keypool_size` could mean that `TopUp()` would generate *more* scriptPubKeys than we need/expect, which would cause migration to assert and crash.

Reply:
Thanks for taking the time to clarify this matter!